cypress/tests/ErrorHandling.js

```diff
@@ -178,3 +181,44 @@ export function ActionUnavailableErrorHandling()
   editDialogModal.ClickSaveCloseButton()
   modelPage.VerifyNoErrorIconOnPage()
 }
+
+Cypress.TestCase('ErrorHandling', 'Missing Action', MissingAction)
+export function MissingAction()
+{
+  models.ImportModel('z-missingAction', 'z-whatsYourName.cl')
+  modelPage.NavigateToTrainDialogs()
+  cy.WaitForTrainingStatusCompleted()
+
+  modelPage.VerifyNoErrorIconOnPage()
+
+  train.CreateNewTrainDialog()
+
+  train.TypeYourMessage(gonnaDeleteAnAction)
+  editDialogModal.ClickScoreActionsButton()
+  train.SelectAction(common.whatsYourName)
+  
+  train.Save()
+
+  modelPage.NavigateToActions()
+  actions.DeleteAction(common.whatsYourName)
+  modelPage.NavigateToTrainDialogs()
+
+  modelPage.VerifyErrorIconForTrainDialogs()
+  train.VerifyErrorsFoundInTraining(`${String.fromCharCode(59412)}${gonnaDeleteAnAction}`, gonnaDeleteAnAction, undefined)
+
+  train.EditTraining(`${String.fromCharCode(59412)}${gonnaDeleteAnAction}`, gonnaDeleteAnAction, undefined)
```

Comment:
Ideally we should find a way to remove the need to do this, but since it's a work around you should at least add a constant to provide semantic meaning to why the code is there.

At the top of this file, or in the common file do something like this:
```
const invalidDialogIcon = String.fromCharCode(59412)
...
train.EditTraining(`${invalidDialogIcon}${gonnaDeleteAnAction}`, gonnaDeleteAnAction, undefined)
```

Now people reading this line in the test what this string is added for because they know what it represents.

Reply:
I would second the use of a constant here

Reply:
Good point...should have thought of that myself. I'll fix it.